### PR TITLE
docs: update examples for tree-shakeable providers

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: aio/content/examples/dependency-injection/src/app/heroes/hero.service.1.2.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// #docregion
```
Comment (gkalpak) [inline aio/content/examples/dependency-injection/src/app/heroes/hero.service.1.2.ts]: What is `hero.service.1.2.ts`??? I m not even sure our infrastructure can handle such names :grin:

File: aio/content/examples/dependency-injection/src/app/heroes/hero.service.ts [hunk 2]
```diff
@@ -2,8 +2,14 @@
...
 import { HEROES }     from './mock-heroes';
 import { Logger }     from '../logger.service';
+import { UserService } from '../user.service';
 
-@Injectable()
+@Injectable({
+  providedIn: 'root',
+  useFactory: (logger: Logger, userService: UserService) =>
+      new HeroService(logger, userService.user.isAuthorized),
+  deps: [Logger, UserService],
```
Comment (gkalpak) [inline aio/content/examples/dependency-injection/src/app/heroes/hero.service.ts]: The corresponding guide section should be updated accordingly: https://pr22961-854a794.ngbuilds.io/guide/dependency-injection#factory-providers

**Review by gkalpak — COMMENTED** [review]
I am not sure what the purpose of the PR is :grin:
The examples look fine (afaict), but the main problems I see are that:

1. The corresponding guides need to be updated to match the new code snippets.
2. There are some files that don't seem to be used for anything.
3. Some files that are no longer used should be removed (e.g. `dependency-injector/src/app/heroes/hero.service.provider.ts`).
4. Several files are included in the live-example/downloadable zip, that may leave the users confused (e.g. what is `HeroesTspComponent`, what is it used for, what does `Tsp` stand for, etc).

##
But yay! for tree-shakeable injectors nonetheless \o/ ❤️

File: aio/content/examples/styleguide/src/07-03/app/app.component.ts [hunk 3]
```diff
@@ -8,6 +8,5 @@ import { HeroService } from './heroes';
   template: `
       <toh-heroes></toh-heroes>
     `,
-  providers: [HeroService]
```
[the remark this answers is not in the document]
Comment (alxhub) [reply aio/content/examples/styleguide/src/07-03/app/app.component.ts]: Guide updates are happening in a separate PR.

File: aio/content/examples/providers/src/app/app.module.ts [hunk 4]
```diff
@@ -6,7 +6,6 @@ import { UserService } from './user.service';
 
 @NgModule({
   imports:      [ BrowserModule ],
-  providers:    [ UserService ],
```
[the remark this answers is not in the document]
Comment (alxhub) [reply aio/content/examples/providers/src/app/app.module.ts]: Guide updates are happening in a separate PR.

File: aio/content/examples/providers/src/app/user.module.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { NgModule } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (alxhub) [reply aio/content/examples/providers/src/app/user.module.ts]: Sorry, I should've given some context. This is the predecessor PR to the docs updates for tree-shakeable providers.

**Comment by alxhub** [comment]
1. Guides will be updated in a future PR.
2. That future PR will use the new examples.
3. This file is used (by `HeroesComponent`).
4. I added a comment to `HeroesTspComponent` explaining what it is.

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**